### PR TITLE
Update submodules and dependencies

### PR DESCRIPTION
**Pull request by tobodner**
This updates our submodules benchmark, cxxopts, googletest, jemalloc, join-order-benchmark, libpgxx, lz4, nlohmann_json, and zstd in order to adopt upstream bug fixes, performance improvements, and features.

The performance in TPC-H with scale factor 1, scheduler and 100 clients changes as follows:
```diff
 +----------------+---------------+-------+---------------+-------+------------+---------+
 | Benchmark      | prev. iter/s  | runs  | new iter/s    | runs  | change [%] | p-value |
 +----------------+---------------+-------+---------------+-------+------------+---------+
-| TPC-H 01       | 2.41641139984 | 145   | 2.23328518867 | 134   | -8%        |  0.2338 |
+| TPC-H 02       | 91.2210464478 | 5474  | 96.0788803101 | 5765  | +5%        |  0.0853 |
 | TPC-H 03       | 36.3489608765 | 2181  | 38.0633773804 | 2284  | +5%        |  0.7103 |
 | TPC-H 04       | 64.0895690918 | 3846  | 61.1996841431 | 3672  | -5%        |  0.4522 |
+| TPC-H 05       | 26.0958499908 | 1566  | 27.5480728149 | 1653  | +6%        |  0.5735 |
+| TPC-H 06       | 1032.89074707 | 10001 | 1080.58496094 | 10002 | +5%        |  0.0000 |
+| TPC-H 07       | 8.79987716675 | 528   | 9.38303279877 | 563   | +7%        |  0.1013 |
+| TPC-H 08       | 46.5157012939 | 2791  | 49.0445861816 | 2943  | +5%        |  0.4014 |
 | TPC-H 09       | 11.3490924835 | 681   | 11.6331157684 | 698   | +3%        |  0.2779 |
 | TPC-H 10       | 21.8970241547 | 1314  | 22.5663070679 | 1354  | +3%        |  0.9260 |
+| TPC-H 11       | 173.771865845 | 10002 | 184.447677612 | 10000 | +6%        |  0.0335 |
 | TPC-H 12       | 95.9611282349 | 5758  | 98.3300094604 | 5900  | +2%        |  0.9218 |
 | TPC-H 13       | 19.0657978058 | 1144  | 19.6165676117 | 1177  | +3%        |  0.4203 |
 | TPC-H 14       | 298.422729492 | 10002 | 300.745758057 | 10001 | +1%        |  0.9706 |
 | TPC-H 15       | 487.218261719 | 10002 | 489.826446533 | 10007 | +1%        |  0.7660 |
 | TPC-H 16       | 67.5801773071 | 4055  | 70.1416931152 | 4209  | +4%        |  0.2059 |
 | TPC-H 17       | 6.71656513214 | 403   | 6.96635770798 | 418   | +4%        |  0.9536 |
+| TPC-H 18       | 3.78312039375 | 227   | 4.13286209106 | 248   | +9%        |  0.8768 |
 | TPC-H 19       | 76.3929595947 | 4584  | 78.2560424805 | 4696  | +2%        |  0.2335 |
 | TPC-H 20       | 15.1146354675 | 907   | 15.0153121948 | 901   | -1%        |  0.1822 |
+| TPC-H 21       | 15.5830421448 | 935   | 16.4829406738 | 989   | +6%        |  0.0153 |
 | TPC-H 22       | 59.7770614624 | 3587  | 61.7823829651 | 3707  | +3%        |  0.6558 |
 | geometric mean |               |       |               |       | +3%        |         |
 +----------------+---------------+-------+---------------+-------+------------+---------+
```

This also drops our dependency on glob2. glob2 is not available for Ubuntu 19.10 yet not required anyways.

### CODE REVIEW

**Comment by mrks** [comment]
> performance improvements

This probably applies mostly to jemalloc. Would be interesting to see a comparison (`--scheduler --clients 100`)

**Comment by Bouncner** [comment]
Sorry. Verklickt.

File: src/benchmark/CMakeLists.txt [hunk 1]
```diff
@@ -30,6 +30,8 @@ target_link_libraries(
     benchmark
 )
 
+target_link_libraries_system(hyriseMicroBenchmarks nlohmann_json::nlohmann_json)
```
Comment (tobodner) [inline src/benchmark/CMakeLists.txt]: Is there a more elegant solution to this problem?

Comment (mrks) [reply src/benchmark/CMakeLists.txt]: ... which is?

**Pull request merged**